src/services/arduinodevconn/app.py

```diff
@@ -0,0 +1,84 @@
+import io
+import logging
+import cherrypy
+import smbus
+
+from common.WIOTRestApp import *
+from common.SettingsManager import *
+from common.SettingsNode import *
+from common.RESTBase import RESTBase
+from common.CatalogRequest import *
+
+class ArduinoDevConnAPI(RESTBase):
+
+    def __init__(self, upperRESTSrvcApp, settings: SettingsNode) -> None:
+        super().__init__(upperRESTSrvcApp, 0)
+        self._catreq = CatalogRequest(self.logger, settings)
+
+        try:
+            self.logger.debug(f"ENV ONRASPBERRY = {str(os.environ['ONRASPBERRY'])}")
+            self._onrpi = str(os.environ["ONRASPBERRY"]) == "1"
+        except KeyError:
+            self._onrpi = False
+
+        if self._onrpi:
+            # setup connection to arduino
+            self._bus = smbus.SMBus(settings.arduino.i2c_dev)
+            self._ard_i2c_addr = int(settings.arduino.i2c_addr, 0)
+
+            pass
+
+        if not self._onrpi:
+            self.logger.warning("Raspberry not found or error as occurred while Arduino init. Running as dummy device!")
+
+    @cherrypy.tools.json_out()
+    def GET(self, *path, **args):
+
+        if len(path) > 0:
+            if path[0] == "switch":
+
+                if not "state" in args:
+                    cherrypy.response.status = 400
+                    return self.asjson_error("Missing state argument")
+
+                state = str(args["state"]).lower()
+                if state != "on" and state != "off":
+                    cherrypy.response.status = 400
+                    return self.asjson_error("Valid states are: {on, off}")
+
+                st = "on" if state == "on" else "off"
```

Comment:
Why do you need this variable st? Couldn't you use directly `state`?

Reply:
Yep if you want I can change it, let me know

Reply:
Yes, I think it is cleaner

Reply:
done https://github.com/thegabriele97/WaterIoT/pull/10/commits/166cbb524d9bfe7519e1e3a2ee10957676d72c02